### PR TITLE
fix(hr): Adjust Page reload for custom frame presenter

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

Fixes Page hot reload on targets which are using native `Frame` presenters, which can cause the Frame to become empty.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by Youssef1313 — APPROVED** [review]

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs [hunk 1]
```diff
@@ -206,6 +206,15 @@ private static void SwapViews(_View oldView, _View newView)
 			{
 				parentAsContentControl.Content = newView;
 			}
+			else if (newView is Page newPage && oldView is Page oldPage)
+			{
+				// In the case of Page, swapping the actual page is not supported, so we
+				// need to swap the content of the page instead. This can happen if the Frame
+				// is using a native presenter which does not use the `Frame.Content` property.
+				oldPage.DataContext = null;
```
Comment (Youssef1313) [inline src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs:214]: Approving, but I'm not sure about this specific line.

**Review by spouliot — APPROVED** [review]
<3 the extra logs

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs:214]: I'm not sure either. We may need to adjust it to propagate the inherited DataContext from the Frame instead, I'll try check for that part and see if it's more appropriate.

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/4.8

**Pull request merged**